### DISCUSSION TITLE
YAML configs | is there a way to isolate external dependencies from commands?

### DISCUSSION THREAD

**Discussion by gofford** [ropensci/targets#601] [Help]
**Is there a way to isolate a shared external file/input from being a dependency for all targets?** 
 
I'm using a YAML config file to configure functions within a pipeline. This is to make the pipeline itself easily configurable, and transportable. Different elements of the config are associated with different targets. But _any_ change to any part of the config file is invalidating all associated targets even if the variable changed in the YAML file is not used by a target. E.g., a target that depends on `config$var1` is still being invalidated by a change to `config$var2` despite the latter not being used by that target. I understand that dependencies are determined by a hash of the source file (and that any change to the config will yield a new hash), but is there a way to configure a target command such that the dependency is only the value used from the config and **not** the global contents of the `config` file? 

This is the scenario I am working with. 

![Screenshot 2021-08-12 at 13 42 24](https://user-images.githubusercontent.com/32452466/129206284-105a0476-aedf-4671-ac42-96b95293f6f0.png)

Every step pulls from a different portion of the `config`. Any change to the config itself invalidates all steps. Ideal solution is that only changes to the variable used by a command invalidate that target, but I haven't yet been able to find a solution. Have you encountered this before?

**Comment by wlandau** [comment]
Maybe split up the config into multiple different files and have each target depend only on the specific file it needs.

**Reply by wlandau** [reply]
Or instead of a target depending on the whole file, create an intermediate target that reads the yaml and just returns a part of it. That would avoid having to awkwardly split up the file. yaml::read_yaml() could help.
